### PR TITLE
sort tweet carousel by relevance all time

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: search/tweets_engine.go [hunk 1]
```diff
@@ -65,17 +67,12 @@ func (e *ESEngine) CombineResultsToSingleHit(resultsByLang map[string]*elastic.S
 	//  Set the score as the highest score of all hits per language.
 
 	for _, result := range resultsByLang {
+		firstHit := result.Hits.Hits[0]
+		source, _ := json.Marshal(es.EffectiveDate{EffectiveDate: &utils.Date{time.Now()}})
```
[the remark this answers is not in the document]
Comment (davgur) [reply search/tweets_engine.go:71]: can't be error here, I make unmarshal from constant

[the remark this answers is not in the document]
Comment (davgur) [reply search/tweets_engine.go]: [commit](https://github.com/Bnei-Baruch/archive-backend/pull/178/commits/7cf82b140ba85c5ea0edcad291d9928ef260afe5)

File: search/tweets_engine.go [hunk 2]
```diff
@@ -86,19 +86,32 @@ func (e *ESEngine) CombineResultsToSingleHit(resultsByLang map[string]*elastic.S
...
 		result.Hits.Hits = []*elastic.SearchHit{hit}
 		result.Hits.TotalHits = 1
-		result.Hits.MaxScore = &maxScore
+		result.Hits.MaxScore = firstHit.Score
 	}
 
 	return resultsByLang, nil
 }
 
+func findFirstHitWithEffectiveDate(hits []*elastic.SearchHit) (*elastic.SearchHit, error) {
```
Comment (davgur) [inline search/tweets_engine.go:103]: @YuriHechter for me it's all time return error (EffectiveDate for all hits are nil)

Comment (YuriHechter) [reply search/tweets_engine.go:103]: even if you sort by time?

Comment (davgur) [reply search/tweets_engine.go:103]: yes, for me it's nil all time

**Pull request closed without merging**